clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -75,28 +85,31 @@ object ElasticsearchIndex extends ModelAutoDerivation {
     DateTimeFormatter.ofPattern("yyyy/MM/dd")
 
   /** Implicit summoner, for convenience. */
-  def apply[Document: ElasticsearchIndex]: ElasticsearchIndex[Document] =
-    implicitly[ElasticsearchIndex[Document]]
+  def apply[Index]: ElasticsearchIndex[Index] =
+    implicitly[ElasticsearchIndex[Index]]
 
-  implicit val WgsUbam: ElasticsearchIndex[DocumentWgsUbam] =
-    new ElasticsearchIndex[DocumentWgsUbam](
+  implicit val WgsUbam: ElasticsearchIndex[WgsUbamIndex.type] =
+    new ElasticsearchIndex(
       "wgs-ubam",
```

Comment:
This is the same as `WgsUbamIndex.commandName`. Can this value be moved to a property of index? We could make `commandName` default to the index name, which seems like what you'd typically want.

Reply:
I'm not sure it'd work in general, because of the -v2 tacked on the end of the gvcf & cram indexes.

Reply:
Even if the names aren't exactly the same, it should be moved to the index class, since it is an index property.

Reply:
I'm not sure which index you mean... `TransferIndex` shouldn't care what the name of the backing ES index is (for ex., whenever we switch all the client-side APIs from "wgs-ubam" to just "ubam" we'll probably keep the ES index named as "wgs-ubam", until we have a good way to migrate cloud storage). This _is_ a property of the `ElasticsearchIndex` class as-is, but it's one that's filled in through a constructor parameter.

Reply:
From a separation of concerns point of view, I agree that they aren't in the same domain. But I think you could imagine `WgsUbamIndex` as a configuration object that represents the superset of all configurable attributes of an index for clio.

Reply:
@pshapiro4broad, are you saying that `"wgs-ubam"`, '"gvcf-v2"`, etc. should be made into members of the `WgsUbamIndex`, `GvcfIndex`, etc. objects, and accessed and passed into the `ElasticsearchIndex` constructor from there?

Reply:
Yes to moving them to members. Once they're members you don't need to pass them anymore because the index is being passed, so `ElasticsearchIndex` code can use the members of the index instead of requiring them as parameters.